### DISCUSSION TITLE
Disabling protected mode (v.7+)

### DISCUSSION THREAD

**Discussion by ilapaylineate** [Grokzen/docker-redis-cluster#149] [Q&A] [answered]
Hello!
Is it possible to disable protected mode for images with 7+ version?
We try to connect to redis from another container/host machine and getting this error:
![image](https://user-images.githubusercontent.com/60742238/208500443-88995b08-a1af-4018-b02b-a59adb4284cb.png)

Setting a password is not possible for us. Can we disable protected mode by setting some flag to false?

**Comment by Grokzen** [comment, marked as answer]
Interesting, this was a new security feature added that i was not aware of in redis 7.0

The short answer is no, there is no feature right now to "simply disable" it via some flag or something. 

But you have a few options to help you out in the short term. One is to modify this file https://github.com/Grokzen/docker-redis-cluster/blob/master/redis-cluster.tmpl to include the config option you want, rebuild the image to you'r own copy and use that one instead. This would map to suggestion `2.` from the above mentioned suggestions by redis. The most simple fix for me to implement would probably be this one out of all the alternatives below.

The second alternative would be to volume map in your own configuration files that you generate separate where you include all the configuration you want for your instances

A third alternative is to fork https://github.com/Grokzen/docker-redis-cluster/blob/master/docker-entrypoint.sh and to modify the entrypoint. It might even be possible to use another entrypoint command all together that you can make before running this script.

A fourth alternative is that after you start your container, you run a `docker exec ...` command that runs the other alternative suggested where you run a `CONFIG SET protected-mode no` to all nodes so it would appear it would be run from inside or from loopback.

A fifth alternative might be that if you run docker-compose solution, there might be options to use different command or entrypoints or to simply just rebuild the image to what you need.
